src/boot.lisp

```diff
@@ -381,6 +381,11 @@
      x)
     ((symbolp x)
      (symbol-function x))
+    ;; This is a HACK that relies on the current storage of setf definitions.
```

Comment:
Nice!  Could we perhaps extract the hack into a function so it's not duplicated between `fdefinition` and `defun`? 

If we ever get rid of the hack, it'd be lot easier to find the implications when changing that function.

Or we might define two `get-setf-handler` and `set-setf-handler`, close to each other, and use them?

Reply:
Sure, the refactoring approach might be better. I can see what I can do about it.

So do you want me to refactor the DEFUN part( https://github.com/jscl-project/jscl/blob/v0.8.2/src/boot.lisp#L110-L139 ) as well? Or maybe just the naming of the symbol here: https://github.com/jscl-project/jscl/blob/v0.8.2/src/boot.lisp#L120-L123

>  If we ever get rid of the hack, it'd be lot easier to find the implications when changing that function.

Since there is a unittest written in this PR we would be able to find any breakages if we ever change the DEFUN part, but I am happy to refactor it.

Reply:
> Since there is a unittest written in this PR we would be able to find any breakages if we ever change the DEFUN part, but I am happy to refactor it.

Right, that's a good point. It doesn't harm if the getter and setter are a bit closer.

But either case, let us know when you are happy with it and we can merge 👍